index_impl.go

```diff
@@ -385,35 +404,27 @@ func memNeededForSearch(req *SearchRequest,
 	// overhead, size in bytes from collector
 	estimate += topnCollector.Size()
 
-	var dm search.DocumentMatch
-	sizeOfDocumentMatch := dm.Size()
-
 	// pre-allocing DocumentMatchPool
-	var sc search.SearchContext
-	estimate += sc.Size() + numDocMatches*sizeOfDocumentMatch
+	estimate += searchContextSizeEstimate + numDocMatches*documentMatchSizeEstimate
 
 	// searcher overhead
 	estimate += searcher.Size()
 
 	// overhead from results, lowestMatchOutsideResults
-	estimate += (numDocMatches + 1) * sizeOfDocumentMatch
+	estimate += (numDocMatches + 1) * documentMatchSizeEstimate
 
 	// additional overhead from SearchResult
-	var sr SearchResult
-	estimate += sr.Size()
+	estimate += reflectStaticSizeSearchResult + reflectStaticSizeSearchStatus
 
 	// overhead from facet results
 	if req.Facets != nil {
-		var fr search.FacetResult
-		estimate += len(req.Facets) * fr.Size()
+		estimate += len(req.Facets) * facetResultSizeEstimate
 	}
 
 	// highlighting, store
-	var d document.Document
 	if len(req.Fields) > 0 || req.Highlight != nil {
-		for i := 0; i < (req.Size + req.From); i++ { // size + from => number of hits
-			estimate += (req.Size + req.From) * d.Size()
-		}
+		// Size + From => number of hits
+		estimate += (req.Size + req.From) * documentSizeEstimate
```

Comment:
the math here seems different than the original -- did the original code have a bug?

Reply:
Yes, it seems like we were double calculating it.